ragna/assistants/_google.py

```diff
@@ -0,0 +1,95 @@
+from typing import Iterator
+
+from ragna.core import PackageRequirement, Requirement, Source
+
+from ._api import ApiAssistant
+
+
+class GoogleApiAssistant(ApiAssistant):
+    _API_KEY_ENV_VAR = "GOOGLE_API_KEY"
+    _MODEL: str
+    _CONTEXT_SIZE: int
+
+    @classmethod
+    def requirements(cls) -> list[Requirement]:
+        return [
+            *super().requirements(),
+            PackageRequirement("json-stream"),
+        ]
+
+    @classmethod
+    def display_name(cls) -> str:
+        return f"Google/{cls._MODEL}"
+
+    @property
+    def max_input_size(self) -> int:
+        return self._CONTEXT_SIZE
+
+    def _instructize_prompt(self, prompt: str, sources: list[Source]) -> str:
+        # https://ai.google.dev/docs/prompt_best_practices#add-contextual-information
+        return "\n".join(
+            [
+                "Answer the prompt using only the pieces of context below.",
+                "If you don't know the answer, just say so. Don't try to make up additional context.",
+                f"Prompt: {prompt}",
+                *[f"\n{source.content}" for source in sources],
+            ]
+        )
+
+    def _call_api(
```

Comment:
This cannot be an async function since the `json_stream` library that we need to parse the response cannot handle async streams.